src/Navigation/Xamarin.Forms/Navigation.Xamarin.Forms/Service/NavigationService.show.cs

```diff
@@ -70,67 +69,49 @@ public virtual async Task ShowAsync<TViewModel>()
         }
 
 
-        private Func<Page, Task> BuildShowAsyncFunc(
-            Page page)
+        public virtual async Task ShowModalAsync<TViewModel>()
+
+            where TViewModel : class, IViewModel
         {
-            if (RootPage is MasterDetailPage &&
-                page is DetailPage)
-            {
-                return SetDetailAsync;
-            }
-            else if (RootPage is TabbedPage &&
-                page is TabPage)
-            {
-                return SetTabAsync;
-            }
-            else return PushAsync;
-        }
+            ThrowIfNotRegistered<TViewModel>();
 
 
-        private async Task PushAsync(
-            Page page)
-        {
-            if (Navigation == null)
-                return;
+            var viewModelInstance =
+                await InitializeViewModelAsync<TViewModel>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5598919)